### PR TITLE
OCPBUGS-36913: openstack: Fix security group tagging

### PR DESCRIPTION
**Pull request by pierreprinetti**
Before this patch, we used the Neutron call to add tags to the newly created security groups. However, that API doesn't accept tags containing special characters such as slash (`/`), even when url-encoded.

With this change, the security groups are tagged with an alternative API call (replace-all-tags) which accepts the tags in a JSON object. Apparently, Neutron accepts special characters (including slash) when they come in a JSON object.

### CODE REVIEW

File: pkg/infrastructure/openstack/preprovision/securitygroups.go [hunk 1]
```diff
@@ -64,7 +64,16 @@ func SecurityGroups(ctx context.Context, installConfig *installconfig.InstallCon
...
 		}
 
-		if err := attributestags.Add(ctx, networkClient, "security-groups", masterGroup.ID, "openshiftClusterID="+infraID).ExtractErr(); err != nil {
+		// The Neutron call to add a tag
+		// (https://docs.openstack.org/api-ref/network/v2/#add-a-tag)
+		// doesn't accept all special characters. Here we use the
+		// "replace-all-tags" call instead, because it accepts a more
+		// robust JSON body.
+		//
+		// see: https://bugzilla.redhat.com/show_bug.cgi?id=2299208
```
Comment (stephenfin) [inline pkg/infrastructure/openstack/preprovision/securitygroups.go:73]: fyi, this is the downstream issue tracker. You should probably link to bug on Launchpad instead https://bugs.launchpad.net/neutron

**Comment by stephenfin** [comment]
CI failures look unrelated

/lgtm
/approve

File: pkg/infrastructure/openstack/preprovision/securitygroups.go (hunk 1 above)
Comment (pierreprinetti) [reply pkg/infrastructure/openstack/preprovision/securitygroups.go:73]: The issue comes from a middleware doing path splitting. I had a chat with the downstream Neutron team after which I wasn't sure whether the middleware was shipped with RHOSP or if it was part of OpenStack itself

**Pull request merged**